.github/workflows/main.yml

```diff
@@ -1,29 +1,39 @@
 on:
   push:
     branches:
-    - master
+      - thesis
   pull_request:
     branches:
-    - master
+      - thesis
 
 jobs:
   typecheck:
+    permissions: write-all
     runs-on: ubuntu-latest
     name: Typechecking
     steps:
-    - name: "Clone repository"
-      uses: actions/checkout@v2
-    - name: Run Agda
-      id: typecheck
-      uses: ayberkt/agda-github-action@v3.1
-      with:
-        main-file: AllModulesIndex.lagda
-        source-dir: source
-        unsafe: true
-    - name: Upload HTML
-      id: html-upload
-      if: github.ref == 'refs/heads/master'
-      uses: actions/upload-artifact@v3
-      with:
-        name: html
-        path: source/html
+      - name: "Clone repository"
+        uses: actions/checkout@v2
+      - name: Run Agda
+        id: typecheck
+        uses: georgejkaye/agda-github-action@v1.5
+        with:
+          main-file: index.lagda.md
+          source-dir: source/TWA/Thesis/
+          unsafe: true
+          css: https://www.cs.bham.ac.uk/~txw467/tychonoff/Agda.css
+      - name: Deploy HTML to branch
+        uses: s0/git-publish-subdir-action@develop
+        if: github.event_name != 'pull_request'
+        env:
+          REPO: self
+          BRANCH: html
+          FOLDER: source/TWA/Thesis/html
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Upload HTML as artifact
+        id: html-upload
+        if: github.event_name == 'pull_request'
+        uses: actions/upload-artifact@v3
+        with:
+          name: html
+          path: source/TWA/Thesis/html
```

Comment:
I’m not sure if it’s a good idea to modify the GitHub Actions workflow. There should be only one global workflow and we should avoid ad-hoc modifications.

I don’t really understand why this change was made and why it was necessary. The part that I get is that @georgejkaye forked the action and added features like using custom CSS, and I guess you wanted to use these features? If I recall correctly, however, he merged these upstream so it should be fine to use the most recent version of upstream instead @georgejkaye’s fork. Can you change this accordingly? We should avoid using a fork.

I guess the rest of the changes are for uploading your development’s HTML as an artefact in an ad-hoc way. Was there a good reason this had to be done this way? In principle, if such modifications are necessary for a contributor’s development, this means our worklow is broken and has to be fixed, which should be reported as an issue.

If you need any help with the action, feel free to make an issue or ask about it on the Discord. I would be happy to generalise the action to accomodate your needs. But I don‘t think we should be making ad-hoc changes without discussion like this one.

Reply:
Everything I changed in the action should have also been merged into ayberkt/agda-github-action now; using my fork was a quick fix to get the html done before the thesis deadline! 

I agree that you should be careful with merging this workflow file though; in particular, the major change is that it pushes the generated html to the 'html' branch, which may or may not be desired. (I also think it's only done for the 'thesis' branch in this workflow from what I remember?